src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -5308,8 +5318,8 @@ internal static List<CompletionResult> CompleteVariable(CompletionContext contex
                         : StringUtil.Format("[{0}]${1}", ToStringCodeMethods.Type(varType, dropNamespaces: true), varName);
 
                     var completionText = !tokenAtCursorUsedBraces && !ContainsCharactersRequiringQuotes(varName)
-                        ? prefix + varName
-                        : prefix + "{" + varName + "}";
+                        ? prefix + scopePrefix + varName
+                        : prefix + "{" + scopePrefix + varName + "}";
```

Comment:
It is not new but have we tests for cases with braces?

Reply:
I see one on line 481.